src/ucs/sys/event_set.c

```diff
@@ -137,44 +140,40 @@ ucs_status_t ucs_event_set_del(ucs_sys_event_set_t *event_set, int event_fd)
 }
 
 ucs_status_t ucs_event_set_wait(ucs_sys_event_set_t *event_set,
-                                unsigned max_events, int timeout_ms,
+                                unsigned *num_events, int timeout_ms,
```

Comment:
why do we need to return num_events?

Reply:
UCT/TCP needs this to do a loop over calling the `ucs_event_set_wait` and calculating remaining events to read

Reply:
maybe we remove this loop from TCP and wait for events just once?

Reply:
there is duplication here - the wait function both calls the callback, and also returns number of events

Reply:
> maybe we remove this loop from TCP and wait for events just once?

TCP has a config parameter `UCX_TCP_MAX_POLL` that can be set to until drained (`-1 = SIZE_MAX`) and we handle it by looping

Do you think that we can remove it? And just wait for `ucs_sys_event_set_max_wait_events` events?

Reply:
i think we can limit it to ucs_sys_event_set_max_wait_events, no need to starve other transports..
@brminich WDYT?

Reply:
or maybe increment the counter by the callback instead of return from wait_for_event (seems cleaner to me)

Reply:
imo, returning num_events looks ok and gives enough flexibility.
Using callback would be more confusing. Do you mean num_events by counter?

Reply:
in tcp, would need 2nd counter..

Reply:
like it is done now with `read_events`? What would callback do?